src/Services/CDADocumentService.php

```diff
@@ -0,0 +1,177 @@
+ */
+
+namespace OpenEMR\Services;
+
+use CouchDB;
+use OpenEMR\Common\Crypto\CryptoGen;
+use OpenEMR\Common\Uuid\UuidRegistry;
+use Symfony\Component\HttpClient\HttpClient;
+
+/**
+ * Class CDADocumentService
+ *
+ * @package OpenEMR\Services
+ *
+ * See interface/modules/zend_modules/module/Carecoordination/src/Carecoordination/Controller/EncounterccdadispatchController.php
+ * indexAction() and interface/modules/zend_modules/public/index.php
+ */
+class CDADocumentService extends BaseService
+{
+    const TABLE_NAME = "ccda";
+    protected $serverUrl;
+
+    public function __construct()
+    {
+        parent::__construct(self::TABLE_NAME);
+        UuidRegistry::createMissingUuidsForTables([self::TABLE_NAME]);
+        $this->serverUrl = $GLOBALS['qualified_site_addr'];
+    }
+
+    public function getLastCdaMeta($pid)
+    {
+        $query = "SELECT cc.uuid, cc.date, pd.fname, pd.lname, pd.pid FROM ccda AS cc
+		    LEFT JOIN patient_data AS pd ON pd.pid=cc.pid
+		    WHERE cc.pid = ?
+		    ORDER BY cc.id DESC LIMIT 1";
+
+        return sqlQuery($query, array($pid));
+    }
+
+    public function getFile($id)
+    {
+        $query = "select couch_docid, couch_revid, ccda_data, encrypted from ccda where uuid=?";
+        $row = sqlQuery($query, array($id));
+        $content = '';
+        if (!empty($row)) {
+            if (!empty($row['couch_docid'])) {
+                $couch = new CouchDB();
+                $resp = $couch->retrieve_doc($row['couch_docid']);
+                if ($row['encrypted']) {
+                    $cryptoGen = new CryptoGen();
+                    $content = $cryptoGen->decryptStandard($resp->data, null, 'database');
+                } else {
+                    $content = base64_decode($resp->data);
+                }
+            } elseif (!empty($row['ccda_data'])) {
+                $fccda = fopen($row['ccda_data'], "r");
+                if ($row['encrypted']) {
+                    $cryptoGen = new CryptoGen();
+                    $content = $cryptoGen->decryptStandard(fread($fccda, filesize($row['ccda_data'])), null, 'database');
+                } else {
+                    $content = fread($fccda, filesize($row['ccda_data']));
+                }
+                fclose($fccda);
+            }
+        }
+
+        return $content;
+    }
+
+    public function generateCCDHtml($pid): string
+    {
+        $url = $this->serverUrl . "/interface/modules/zend_modules/public/encounterccdadispatch";
+        $httpClient = HttpClient::create([
+            "verify_peer" => false,
+            "verify_host" => false
+        ]);
+        $response = $httpClient->request('GET', $url, [
+            'query' => [
+                'combination' => $pid,
+                'recipient' => 'self',
+                'view' => '1',
+                'site' => $_SESSION ['site_id'],
+                'sent_by_app' => 'core_api',
+                'me' => session_id(),
+                'XDEBUG_SESSION' => '1'
```

Comment:
looks like `XDEBUG_SESSION` has weaseled itself back into the codebase :)

Reply:
![make-it-look-easy](https://user-images.githubusercontent.com/5618372/126378286-42ebcf5d-7084-4096-9f44-7cf724f0d277.gif)

Reply:
i'll remove it on my pr :)